WooCommerce/Classes/POS/Controllers/PointOfSaleOrderController.swift

```diff
@@ -110,29 +106,23 @@ protocol PointOfSaleOrderControllerProtocol {
 
     @MainActor
     func sendReceipt(recipientEmail: String) async throws {
-        var isEligibleForPOSReceipt: Bool?
+        var isEligibleForPOSReceipt: Bool = false
```

Comment:
AFAU we have no reason for this to be nil, since it's either eligible or it isn't.  Also allows to simplify the value we pass to the catch block and the track event.

Reply:
Previously, if there was no order, or `orderService.updatePOSOrder` threw an error, we'd track the `receiptEmailFailed` event without `eligible_for_pos_receipt` (thanks to the compactMapValues.)

Now we'll log it as false even if the real answer would have been true.

Perhaps we should just set the `isEligible` at the top instead of after updating the order. The only thing it uses is the siteID, which isn't going to change from the order update, surely.